src/incubator/TouchableOpacity2.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {PropsWithChildren, useCallback, useMemo} from 'react';
+import {LayoutChangeEvent} from 'react-native';
+import Reanimated, {
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+  interpolate,
+  interpolateColor,
+  runOnJS
+} from 'react-native-reanimated';
+import {TapGestureHandler, LongPressGestureHandler, State} from 'react-native-gesture-handler';
+import {asBaseComponent, forwardRef, BaseComponentInjectedProps, ForwardRefInjectedProps} from '../commons/new';
+import {ViewProps} from '../components/view';
+
+export type TouchableOpacityProps = {
+  /**
+   * Background color
+   */
+  backgroundColor?: string;
+  /**
+   * Background color when actively pressing the touchable
+   */
+  feedbackColor?: string;
+  /**
+   * Opacity value when actively pressing the touchable
+   */
+  activeOpacity?: number;
+  /**
+   * Scale value when actively pressing the touchable
+   */
+  activeScale?: number;
+  /**
+   * Callback for when tapping the touchable
+   */
+  onPress?: (props: any) => void;
+  /**
+   * Callback for when long pressing the touchable
+   */
+  onLongPress?: (props: any) => void;
+  /**
+   * Pass controlled pressState to track gesture state changes
+   */
+  pressState?: State;
+  /**
+   * If true, disable all interactions for this component.
+   */
+  disabled?: boolean;
+  /**
+   * Pass custom style
+   */
+  style?: ViewProps['style'];
+  onLayout?: (event: LayoutChangeEvent) => void;
+  testID?: string;
+};
+
+type Props = PropsWithChildren<TouchableOpacityProps & BaseComponentInjectedProps & ForwardRefInjectedProps>;
+
+/**
+ * @description: a Better, enhanced TouchableOpacity component
+ * @modifiers: flex, margin, padding, background
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/incubatorScreens/TouchableOpacityScreen.js
+ */
+function TouchableOpacity(props: Props) {
+  const {
+    children,
+    modifiers,
+    style,
+    // onPress = _.noop,
+    // onLongPress,
```

Comment:
I guess we can remove these two props from here

Reply:
Removed (: